### PR TITLE
UAF-82 ERE Changes - Direct Bill Project for Graduate Tuition Epic Branch

### PR DESCRIPTION
**Pull request by mcalopez**
All code features have been merged into Epic branch.
* UAF-3230

### CODE REVIEW

File: kfs-ld/src/main/java/edu/arizona/kfs/module/ld/LaborConstants.java [hunk 1]
```diff
@@ -25,4 +25,7 @@
 	
 	public static final String[] BALANCE_TYPES = {"AC", "CB", "IE"};
 
+	//SalaryExpenseTransferDocument
+	public static final String BALANCE_TYPE_ACTUAL = "AC";
+	public static final String BALANCE_TYPE_A2 = "A2";
```
Comment (ghost) [inline kfs-ld/src/main/java/edu/arizona/kfs/module/ld/LaborConstants.java:30]: You should really be using System Options to get balance type codes as these could change by the fiscal year (OK, UA values have never changed). So, for the future, this is how it should be done:

`//get current year system options
SystemOptions options = getSystemOptions(universityDateService.getCurrentFiscalYear());

// set to AC
e.setFinancialBalanceTypeCode(options.getActualFinancialBalanceTypeCd());

public SystemOptions getSystemOptions(Integer universityFiscalYear) {
    SystemOptions options = null;
    options = getOptionsService().getOptions(universityFiscalYear);
    if (ObjectUtils.isNull(options)) {
        options = getOptionsService().getCurrentYearOptions();
    } 
    return options;
}`

File: kfs-ld/src/main/java/edu/arizona/kfs/module/ld/document/SalaryExpenseTransferDocument.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+		BigDecimal fringeBenefitPercent = tuitionRemissionAmount.divide(salaryAmount, 12, BigDecimal.ROUND_HALF_UP);
+		return new KualiDecimal(fringeBenefitPercent.multiply(expenseTransferAccountingLine.getAmount().bigDecimalValue()));
+	}
+
+	public BigDecimal amountForCalculation(ExpenseTransferAccountingLine expenseTransferAccountingLine, String objectCode) {
+		Map<String, String> fieldValues = new HashMap<String, String>();
+		fieldValues.put(KFSPropertyConstants.UNIVERSITY_FISCAL_YEAR, expenseTransferAccountingLine.getPayrollEndDateFiscalYear().toString());
+		fieldValues.put(KFSPropertyConstants.EMPLID, expenseTransferAccountingLine.getEmplid());
+		fieldValues.put(KFSPropertyConstants.FINANCIAL_OBJECT_CODE, objectCode);
```
Comment (ghost) [inline kfs-ld/src/main/java/edu/arizona/kfs/module/ld/document/SalaryExpenseTransferDocument.java]: Yikes, this lookup needs to include position number (see KATTS-3656). You've no idea what trouble this caused when this mod originally went live (it was a UA president requested mod).

Comment (mcalopez) [reply kfs-ld/src/main/java/edu/arizona/kfs/module/ld/document/SalaryExpenseTransferDocument.java]: I will work on this as soon as possible.

File: kfs-ld/src/main/java/edu/arizona/kfs/module/ld/LaborConstants.java (hunk 1 above)
Comment (mcalopez) [reply kfs-ld/src/main/java/edu/arizona/kfs/module/ld/LaborConstants.java:30]: I have taken note of this

**Comment by ghost** [comment]
Please see my comments (especially the second one). Otherwise, looks good.

File: kfs-ld/src/main/java/edu/arizona/kfs/module/ld/document/SalaryExpenseTransferDocument.java (hunk 2 above)
Comment (mcalopez) [reply kfs-ld/src/main/java/edu/arizona/kfs/module/ld/document/SalaryExpenseTransferDocument.java]: Position number was included on this look up on the new commit

File: kfs-ld/src/main/java/edu/arizona/kfs/module/ld/LaborConstants.java (hunk 1 above)
Comment (mcalopez) [reply kfs-ld/src/main/java/edu/arizona/kfs/module/ld/LaborConstants.java:30]: This change was implemented. on the new commit. However,  I noticed that code A2 does not exist on the System Options table (FS_OPTION_T), so I kept the constant for Balance Type A2.

Comment (ghost) [reply kfs-ld/src/main/java/edu/arizona/kfs/module/ld/LaborConstants.java:30]: Mea culpa, I didn't realize that A2  balance type code was not available on the Systems Options table. I am now OK with the use of these two balance type code constants. As we need to remove your inclusion of OriginEntryFull BO in SalaryExpenseTransferDocument.java anyway, lets go ahead and revert to using the constants.

Comment (ghost) [reply kfs-ld/src/main/java/edu/arizona/kfs/module/ld/LaborConstants.java:30]: Please remove unneeded imports and properties for OriginEntryFull, OptionsService, UniversityDateService, OptionsServiceImpl, UniversityDateServiceImpl from SalaryExpenseTransferDocument.java.

**Comment by ghost** [comment]
Looks good! Good job!

**Pull request merged**